resources/Microsoft.VSCode.DSC/Microsoft.VSCode.DSC.psm1

```diff
@@ -0,0 +1,125 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+$ErrorActionPreference = "Stop"
+Set-StrictMode -Version Latest
+
+enum Ensure
+{
+    Absent
+    Present
+}
+
+#region DSCResources
+[DSCResource()]
+class VSCodeExtension
+{
+    [DscProperty(Key)]
+    [string] $Name
+
+    [DscProperty()]
+    [string] $Version
+
+    [DscProperty()]
+    [Ensure] $Ensure = [Ensure]::Present
+
+    [DscProperty(NotConfigurable)]
+    [hashtable] $InstalledExtensions = @{}
+
+    [VSCodeExtension] Get()
+    {
+        Assert-VSCode
+
+        $currentState = [VSCodeExtension]::new()
+        $currentState.Ensure = [Ensure]::Absent
+        $currentState.Name = $this.Name
```

Comment:
I preferred using the value returned by the underlying data store when it was found, mostly for casing.

Reply:
Value is now updated with underlying data store if extension is found.